pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3870,6 +3879,27 @@ private void unfenceTopicToResume() {
         subscriptions.values().forEach(sub -> sub.resumeAfterFence());
         isFenced = false;
         isClosingOrDeleting = false;
+        replicatorsResume();
+    }
+
+    private void replicatorsResume() {
+        removeTerminatingReplicators(replicators);
+        removeTerminatingReplicators(shadowReplicators);
+        checkReplication();
+        checkShadowReplication();
+    }
+
+    private static void removeTerminatingReplicators(ConcurrentOpenHashMap<String, Replicator> replicators) {
```

Comment:
Maybe this method doesn't need the `static` qualifier.

Reply:
For performance and test, static method is better 😘